src/import/loadConfigurationFromIgResource.ts

```diff
@@ -11,19 +11,30 @@ import { logger } from '../utils';
  * @param input - path to the input directory
  * @returns {Configuration} the extracted configuration
  */
-export function loadConfigurationFromIgResource(input: string): Configuration | null {
+export function loadConfigurationFromIgResource(
+  input: string,
+  isLegacyIgPubContext: boolean
+): Configuration | null {
   // First, look in the ig.ini file for a path to the IG resource
   let igPath: string;
-  const igIniPath = path.join(input, '..', 'ig.ini');
+  const igIniPath = isLegacyIgPubContext
+    ? path.join(input, '..', 'ig.ini')
+    : path.join(input, 'ig.ini');
```

Comment:
It seems like you have this pattern a few times:
```
isLegacyIgPubContext ? path.join(input, '..', 'ig.ini') : path.join(input, 'ig.ini');
```
Would it work to just do something like that once to find the root directory (the one containing ig.ini), and use that everywhere?

Reply:
Yeah, OK.  I thought about that at first but thought (at that point) there were only two places where this happened.  Then in testing I found I missed one and did not revisit the decision.  I'll refactor.